### PR TITLE
chore: bump celestia-app to v0.12.0-rc5

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

bumps to v0.12.0-rc5

fwiw, I ran into a kinda funny issue since the imports in the proto files changed. We can't regenerate the proto files until we call `go mod tidy`, but we can't call `go mod tidy` until we regenerate the proto files. For posterity, the only way to fix this is to manually edit the proto files to a point where we can call `go mod tidy`, then regenerate the protofiles.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -322,6 +322,7 @@ require (
 )
 
 replace (
+	github.com/celestiaorg/nmt => github.com/celestiaorg/nmt v0.12.0 // temporary patch until we either fix nmt v0.13.0 or downgrade across all repos
```
Comment (evan-forbes) [inline go.mod:325]: as mentioned in the comment, we can either downgrade across all repos, or simply remove this after we fix whatever is breaking v0.13.0

Comment (Wondertan) [reply go.mod]: We decided to remove replace as we fixed the issue with nmt and updated it in https://github.com/celestiaorg/celestia-node/pull/1651

**Comment by evan-forbes** [comment]
whoops lol will go fix

**Comment by Wondertan** [comment]
It's fixed

**Comment by Wondertan** [comment]
But units are failing though

**Comment by Wondertan** [comment]
Ok, it's a know flaky one, ready to go then

**Comment by evan-forbes** [comment]
I can go debug, it looks like `TestRetriever_ByzantineError` is failing on the CI, but this test passes locally for me

**Comment by Wondertan** [comment]
> it's a know flaky one

**Comment by evan-forbes** [comment]
thanks for fixing! sorry about the silly mistakes

**Review by renaynay — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**